src/Func.h

```diff
@@ -344,6 +347,71 @@ class FuncRef {
     EXPORT Internal::Function function() const {return func;}
 };
 
+/** A fragment of front-end syntax of the form f(x, y, z)[index], where x, y,
+ * z are Vars or Exprs. If could be the left hand side of an update
+ * definition, or it could be a call to a function. We don't know
+ * until we see how this object gets used.
+ */
+class FuncTupleElementRef {
+    FuncRef func_ref;
+    std::vector<Expr> args; // args to the function
+    int idx;                // Index to function outputs
+
+    /** Helper function that generates a Tuple where element at 'idx' is set
+     * to 'e' and the rests are undef. */
+    Tuple values_with_undefs(Expr e) const;
+
+public:
+    FuncTupleElementRef(const FuncRef &ref, const std::vector<Expr>& args, int idx);
+
+    /** Use this as the left-hand-side of an update definition of a Func with
+     * multiple outputs indexed by 'idx' (see \ref RDom). The function must
+     * already have an initial definition.
+     */
+    EXPORT Stage operator=(Expr e);
+
+    /** Define this function with multiple outputs indexed by 'idx' as a sum
```

Comment:
Suggested phrasing:

Define Tuple component 'idx' of this Func as a sum reduction over the given expression. The other Tuple components are unchanged. The expression may refer to some RDom to sum over. The function must already have an initial definition.